src/Algebra/Lattice/Structures.agda

```diff
@@ -0,0 +1,106 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Some lattice-like structures defined by properties of _∧_ and _∨_
+-- (not packed up with sets, operations, etc.)
+------------------------------------------------------------------------
+
+-- The contents of this module should be accessed via `Algebra.Lattice`,
+-- unless you want to parameterise it via the equality relation.
+
+{-# OPTIONS --without-K --safe #-}
+
+open import Relation.Binary using (Rel; Setoid; IsEquivalence)
+
+module Algebra.Lattice.Structures
+  {a ℓ} {A : Set a}  -- The underlying set
+  (_≈_ : Rel A ℓ)    -- The underlying equality relation
+  where
+
+open import Algebra.Core
+open import Algebra.Definitions _≈_
+open import Algebra.Structures _≈_
+open import Level using (_⊔_)
+
+
+record IsSemilattice (∙ : Op₂ A) : Set (a ⊔ ℓ) where
+  field
+    isBand : IsBand ∙
+    comm   : Commutative ∙
+
+  open IsBand isBand public
+
+module IsMeetSemilattice {∧} (isSemilattice : IsSemilattice ∧) where
+  open IsSemilattice isSemilattice public
+    renaming (∙-cong to ∧-cong; ∙-congˡ to ∧-congˡ; ∙-congʳ to ∧-congʳ)
+
+module IsJoinSemilattice {∨} (isSemilattice : IsSemilattice ∨) where
+  open IsSemilattice isSemilattice public
+    renaming (∙-cong to ∨-cong; ∙-congˡ to ∨-congˡ; ∙-congʳ to ∨-congʳ)
+
+
+IsBoundedSemilattice = IsIdempotentCommutativeMonoid
```

Comment:
A comment above here, saying that a bounded semilattice is the same thing as an idempotent commutative monoid would be useful!